### ISSUE TITLE
[Feature Request] Disable Upload Entirely by default / move this setting to main settings window

### ISSUE THREAD

**Issue by igroglaz** [ShareX/ShareX#6561]
Right now new users might be in big trouble when they install this program and make screenshot of some private stuff, like CC number, etc - when it will  be uploaded accidentally to remote site. Dangerous default behaviour.

So based on this ticket: https://github.com/ShareX/ShareX/issues/4074
Also people mentioned it https://twitter.com/sharex/status/1218685133894639616?lang=en and in other places.

Uploading should be disabled by default. Or at least this setting must be prompt at program installation.

**Comment by jerbasco1** [comment]
Agreed, this is dangerous default behavior.

**Comment by QuantumGrizzly** [comment]
+1 I got burned by reinstalling a newer version of ShareX on a computer that behaved differently that the older version I used for a long time. I realized after a few weeks that ShareX uploaded hundreds of screenshots on Imgur. That was a nightmare.

The official [ShareX Twitter account](https://twitter.com/ShareX/status/1400777894113202184?s=20&t=IKeYhriAqudG7OR9e10TxQ) mentioned that the majority of users **want** to upload their screenshots on Imgur. If I am being honest, I am a little skeptical of that claim. Is there any evidence (survey, telemetry data, UX sessions) to back that assertion?

**Comment by mmoore59** [comment]
I agree with this suggestion and all the comments.  Am about to recommend this tool to my colleagues, but am first writing a guide to ensure they turn OFF the automatic upload as it could give some security and privacy issues.

**Comment by Jones-Bones** [comment]
Default "after capture" steps are beyond insane. Even with those tweaked, upload should definitely be turned off by default. I have accidentally uploaded screenshots to imgur multiple times since I often use copy to clipboard button from Editor and upload button is located right next to that, finally disabled upload completely today. At least there is an easy way to delete from imgur...

**Comment by savaloi** [comment]
+1 agree with all the above, this is a crazy default setting - the default should 100% be not to upload anything. Fair enough to add the functionality for those who want it, but it should be opt-in, not opt-out.

I would like to see 4 changes implemented:
1. Setup option asking users to tick a box (unticked by default), if they want the upload feature enabled. Ensure this option is shown in Upgrade installs too.
2. CLI switch to allow scripted deployment of ShareX to disable Upload - EG "\DisableUpload=FALSE" or the like.
3. If the upload feature is enabled (via Setup or CLI), users should be prompted each time they take a screenshot or video asking if they want it uploaded or not - it should not be automatic. Again, this should be an opt-in option, not opt-out or automatic.
4. This option should be moved from 'Advanced' to somewhere much more visible in the GUI settings.

**Comment by Alaponuke** [comment]
is there any way to remove the upload button?
![07-02-2023_12-39-26](https://user-images.githubusercontent.com/92026112/217235916-30b92bb7-d65a-46de-820b-1b21270433dd.png)

**Comment by MartenBE** [comment]
I just tried ShareX as a replacement for GreenShot and was suprised to see my screenshot on imgur. This is a big no no! It broke my trust and I am looking for another alternative

**Comment by igorhbs** [comment]
This is insane! Even worse that clicking No the first time the alert pops up still do not globally disable it.

**Comment by 7368697661** [comment]
this made me sign back in because this is a massive security flaw. upload should absolutely be opt in, not opt out, and should be disabled by default. not only that, the option to actually disable upload should not require a google search and reading github. i really cannot in good faith recommend this program to anyone despite fantastic features because of this.

**Comment by pfgrobler-work** [comment]
Having Default file upload enabled makes it impossible to get InfoSec Approval an many corporates environments

**Comment by intensite** [comment]
Is it possible to fork the project and have all network access removed ?

**Comment by Hammerfest** [comment]
Adding my vote for disabled/local only as a default for NEW installs at least, I recommended this to someone and they notified me of its default.

Was recommended due to my prior use with another service (of which I knew it was uploading online), and Snipping Tool's deprecation and the replacement form Microsoft being less then desirable.

**Comment by v1ckxy** [comment]
+1. No "automatic uploads" option should be enabled by default.
It makes no sense and makes the application insecure by default.

**Comment by MojoMoe** [comment]
Yup, I second this, please DISABLE uploads by default, it should be an opt in if someone wants it:

+1. No "automatic uploads" option should be enabled by default.
It makes no sense and makes the application insecure by default.

**Comment by nodejsg** [comment]
I second this, No "automatic uploads" option should be enabled by default. Please DISABLE uploads by default, it's crazy to upload my private pictures!
There is a NO UPLOAD version. https://github.com/Cosebdd/ShareNot

**Comment by 2mustange** [comment]
There was a merge that got ignored #7887 

I find it funny how there are major concerns with automatic upload being default enabled. So much unnecessary risk.

**Comment by BrycensRanch** [comment]
Fixed by https://github.com/ShareX/ShareX/commit/8361e971e23f25d9157a51ab34296e10877a613f

**Comment by carpentern** [comment]
I'm trying to find the commit # for this feature request was implemented so I can reference it. How can I find it, or does anyone know?

**Comment by 2mustange** [comment]
> I'm trying to find the commit # for this feature request was implemented so I can reference it. How can I find it, or does anyone know?

@carpentern its literally posted right above your comment. But the 19.01 release notes also reference it
